08_education/digital_access.qmd

```diff
@@ -0,0 +1,939 @@
+    # Digital variables
+    "total_households_digital" = "B28003_004"
+  )
+
+# Apply our function to multiple datasets using map2 and assigns the results
+# back to the specified object names in the global environment, reducing repetitive code such as:
+# county_overall <- pull_census_data(geography = "county",
+#                                    vars = overall_vars,
+#                                    years = acs1_years,
+#                                    survey = "acs1")
+# place_overall <- pull_census_data(geography = "place",
+#                                    vars = overall_vars,
+#                                    years = acs1_years,
+#                                    survey = "acs1")
+map(
+  .x = c("county", "place"),
+  .f = ~ assign(
+    x = paste0(.x, "_overall"),
+    value = pull_census_data(
+      geography = .x,
+      vars = overall_vars,
+      years = acs1_years,
+      survey = "acs1"
+    ),
+    envir = .GlobalEnv
+  )
+)
+
+county_overall
+place_overall
+```
+
+### Race/ethnicity subgroup
+
+-   Unit of analysis: Population in households,
+-   Survey: ACS 5
+-   Census table(s): B28009, S2802 (subject table)
+-   Years: 2018, 2021
+
+```{r}
+acs5_years <- c(2018, 2021)
+
+raceeth_vars <-
+  c( # Population variables
+    "total_people_in_households" = "S2802_C01_001",
+    "white" = "B28009A_001",
+    "black" = "B28009B_001",
+    "aian" = "B28009C_001",
+    "asian" = "B28009D_001",
+    "nhpi" = "B28009E_001",
+    "other" = "B28009F_001",
+    "two_or_more" = "B28009G_001",
+    "hispanic" = "B28009I_001",
+    # Digital variables
+    "total_people_in_households_digital" = "S2802_C02_001",
+    "white_digital" = "S2802_C02_005",
+    "black_digital" = "S2802_C02_006",
+    "aian_digital" = "S2802_C02_007",
+    "asian_digital" = "S2802_C02_008",
+    "nhpi_digital" = "S2802_C02_009",
+    "other_digital" = "S2802_C02_010",
+    "two_or_more_digital" = "S2802_C02_011",
+    "hispanic_digital" = "S2802_C02_012"
+  )
+
+map(
+  .x = c("county", "place"),
+  .f = ~ assign(
+    x = paste0(.x, "_raceeth"),
+    value = pull_census_data(
+      geography = .x,
+      vars = raceeth_vars,
+      years = acs5_years,
+      survey = "acs5"
+    ),
+    envir = .GlobalEnv
+  )
+)
+
+county_raceeth
+place_raceeth
+```
+
+We aggregate the race/ethnicity categories into Other races and ethnicities,
+Black non-Hispanic, Hispanic, White non-Hispanic, and All (total population in
+households)
+
+```{r}
+#| label: aggregate-raceeth
+aggregate_raceeth <- function(data) {
+  data |>
+    mutate(
+      other = asian + aian + nhpi + other + two_or_more
+    ) |>
+    mutate(
+      other_digital = asian_digital + aian_digital + nhpi_digital + other_digital + two_or_more_digital
+    ) |>
+    mutate(
+      moe_other_digital = sqrt(
+        moe_aian_digital^2 +
+          moe_asian_digital^2 +
+          moe_nhpi_digital^2 +
+          moe_other_digital^2 +
+          moe_two_or_more_digital^2
+      )
```

Comment:
I don't dispute this square root method is valid, but would maybe add a line of explanation for this MOE aggregation process

Reply:
Ah, I see you link out to something later in the code -- if that's the same rationale, maybe just point to it

Reply:
Moved this up so folks can see it the first time I do this aggregation!